src/jit/morph.cpp

```diff
@@ -83,6 +83,18 @@ GenTreePtr          Compiler::fgMorphIntoHelperCall(GenTreePtr      tree,
     tree->gtCall.gtEntryPoint.addr = nullptr;
 #endif
 
+#if defined(_TARGET_X86_) && !defined(LEGACY_BACKEND)
+    if (varTypeIsLong(tree))
+    {
+        GenTreeCall* callNode = tree->AsCall();
+        ReturnTypeDesc* retTypeDesc = callNode->GetReturnTypeDesc();
+        retTypeDesc->Reset();
+        retTypeDesc->Initialize(this, callNode->gtRetClsHnd);
+        
+        NYI("Helper with TYP_LONG return type");
+    }
+#endif
+
```

Comment:
As far as I can tell, no, the tree does not have a GT_STOREIND. What seems to be happening is that we decompose the args to the call, then we decompose the call to be var = call(), but then we try to decompose one of the args again. I'm not sure why it's doing this, and why it only tries to do this with a node that has been morphed into a call (is there some strange ordering of events that happens that I'm not aware of?).